File: src/application/useCases/submitAssessment.ts
import { AssessmentRepository } from '@domain/repositories/assessmentRepository';
import { AssessmentResult } from '@domain/entities/assessmentResult';

export const submitAssessment = async (
  repository: AssessmentRepository,
  patientId: string,
  scaleId: string,
  responses: Record<string, number | boolean | string>
): Promise<AssessmentResult> => {
  return repository.submitAssessment(patientId, scaleId, responses);
};

File: lib/domain/usecases/submit_assessment.dart
import '../entities/assessment_result.dart';
import '../repositories/assessment_repository.dart';

class SubmitAssessment {
  const SubmitAssessment(this._repository);

  final AssessmentRepository _repository;

  Future<AssessmentResult> call({
    required String patientId,
    required String scaleId,
    required Map<String, int> answers,
  }) {
    return _repository.submitAssessment(
      patientId: patientId,
      scaleId: scaleId,
      answers: answers,
    );
  }
}
